src/main/java/me/botsko/prism/utils/ItemUtils.java

```diff
@@ -22,506 +18,559 @@
+     * @return bool
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static boolean isValidItem(ItemStack item) {
+        return (item != null && !item.getType().equals(Material.AIR));
+    }
+
+    /**
+     * Check ItemStack are the same Type.
+     * @param a ItemStack
+     * @param b ItemStack
+     * @return bool
+     */
+    public static boolean isSameType(ItemStack a, ItemStack b) {
+        return a.getType().equals(b.getType());
+    }
+
+    /**
+     * Check if an ItemStack is equal.
+     * 1. Type
+     * 2. Display name
+     * 3. If leather check the colour.
+     * 4. Lore
+     * 5. Enchants.
+     * 6. SkullMeta is checked if appropriate.
+     * 7. Bookmeta - title , page count and author
+     * 8. PotionMeta - checked
+     * 9 FireworkMeta including effect meta.
+     * @param a Itemstack
+     * @param b ItemStack
+     * @return bool
+     */
+    public static boolean equals(ItemStack a, ItemStack b) {
+
+        ItemMeta metaA = a.getItemMeta();
+        ItemMeta metaB = b.getItemMeta();
+
+        // Type/dura
+        if (!isSameType(a, b)) {
+            return false;
+        }
+        // Display name
+        if (metaA.getDisplayName() != null) {
+            if (!metaA.getDisplayName().equals(metaB.getDisplayName()))
+                return false;
+        } else {
+            if (metaB.getDisplayName() != null) {
+                return false;
+            }
+        }
+
+        // Coloring
+        if (metaA instanceof LeatherArmorMeta) {
+            if (!(metaB instanceof LeatherArmorMeta)) {
+                return false;
+            }
+            LeatherArmorMeta colorA = (LeatherArmorMeta) metaA;
+            LeatherArmorMeta colorB = (LeatherArmorMeta) metaB;
+            if (!colorA.getColor().equals(colorB.getColor())) {
+                return false;
+            }
+        }
+
+        // Lore
+        if (metaA.getLore() != null && metaA.getLore() != null) {
+            for (String lore : metaA.getLore()) {
+                if (!Objects.requireNonNull(metaB.getLore()).contains(lore)) {
+                    return false;
+                }
+            }
+        } else if (!(metaA.getLore() == null && metaB.getLore() == null)) {
+            return false;
+        }
+
+        // Enchants
+        if (enchantsUnEqual(a.getEnchantments(), b.getEnchantments())) {
+            return false;
+        }
+
+        // Books
+        if (metaA instanceof BookMeta) {
+            if (!(metaB instanceof BookMeta)) {
+                return false;
+            }
+
+            BookMeta bookA = (BookMeta) metaA;
+            BookMeta bookB = (BookMeta) metaB;
+
+            // Author
+            if (bookA.getAuthor() != null) {
+                if (!bookA.getAuthor().equals(bookB.getAuthor())) {
+                    return false;
+                }
+            }
+
+            if (bookA.getTitle() != null) {
+                if (!bookA.getTitle().equals(bookB.getTitle())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)